packages/compass-databases-navigation/src/icon-button.tsx

```diff
@@ -6,22 +6,46 @@ import {
   spacing,
   css,
   cx,
-  uiColors,
 } from '@mongodb-js/compass-components';
+import {
+  darkIconColor,
+  darkIconColorHover,
+  lightIconColor,
+  lightIconColorHover,
+} from './constants';
 
 const iconContainer = css({
-  color: uiColors.white,
   display: 'block',
   flex: 'none',
   fontSize: 0,
 });
 
+export type IconMode = 'normal' | 'hovered';
+
+function pickColor(darkMode: boolean, mode: IconMode) {
+  if (darkMode) {
+    if (mode === 'hovered') {
+      return darkIconColorHover;
+    }
+    return darkIconColor;
+  } else {
+    if (mode === 'hovered') {
+      return lightIconColorHover;
+    }
+    return lightIconColor;
+  }
+}
+
 export const SmallIcon: React.FunctionComponent<
-  { glyph: string } & React.HTMLProps<HTMLSpanElement>
-> = ({ glyph, className, ...props }) => {
+  { glyph: string; mode: IconMode } & React.HTMLProps<HTMLSpanElement>
+> = ({ glyph, mode, className, ...props }) => {
+  const useNewSidebar = process?.env?.COMPASS_SHOW_NEW_SIDEBAR === 'true';
+  const darkMode = !useNewSidebar; // for now assume the old sidebar is dark and the new one light
```

Comment:
My guess is that once we properly implement darkmode and the both the connetions and normal sidebars are light when darkmode is off and dark when it is on then we'd be pulling darkMode out of a context or something as low as we need it and not pass it down as a prop?

Reply:
IIRC we already have a dark mode context and `useDarkMode` hook that you should be able to use here

Reply:
Ah, [it's called `useTheme`](https://github.com/mongodb-js/compass/blob/main/packages/compass-components/src/hooks/use-theme.tsx)

Reply:
Nice I'll look into it. Although.. right now we're in this weird state where there is only a dark version of the connections sidebar regardless of whether were in dark mode or not. Which complicates matters a bit.